bot/cogs/webhook_remover.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import re
+import typing as t
+
+from discord import Colour, Message
+from discord.ext.commands import Cog
+
+from bot.bot import Bot
+from bot.cogs.moderation.modlog import ModLog
+from bot.constants import Channels, Colours, Event, Icons
+
+WEBHOOK_URL_RE = re.compile(r"discordapp\.com/api/webhooks/\d+/\S+/?")
+
+ALERT_MESSAGE_TEMPLATE = (
+    "{user}, looks like you posted Discord Webhook URL to chat. "
+    "I removed this, but we **strongly** suggest to change this now "
+    "to prevent any spam abuse to channel. Please avoid doing this in future. "
+    "If you believe this was mistake, please let us know."
+)
+
+log = logging.getLogger(__name__)
+
+
+class WebhookRemover(Cog):
+    """Scan messages to detect Discord webhooks links."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @property
+    def mod_log(self) -> ModLog:
+        """Get current instance of `ModLog`."""
+        return self.bot.get_cog("ModLog")
+
+    async def scan_message(self, msg: Message) -> t.Tuple[bool, t.Optional[str]]:
+        """Scan message content to detect Webhook URLs. Return `bool` about does this have Discord webhook URL."""
+        matches = WEBHOOK_URL_RE.search(msg.content)
+        if matches:
+            return True, matches[0]
+        else:
+            return False, None
+
+    async def delete_and_respond(self, msg: Message, url: str) -> None:
+        """Delete message and show warning when message contains Discord Webhook URL."""
+        # Create URL that will be sent to logs, remove token
+        parts = url.split("/")
+        parts[-1] = "xxx"
+        url = "/".join(parts)
```

Comment:
This parsing may be simplified by adding capture groups to the initial regex and using those groups.